blocks/counter/index.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/**
+ * Block Name: Counter.
+ *
+ * @package wp-curate
+ *
+ * phpcs:disable Squiz.Commenting.FunctionComment.MissingParamTag
+ * phpcs:disable Squiz.Commenting.FunctionComment.MissingParamName
+ */
+
+use WP_Curate\Core\Block;
+
+use function Alley\WP\add_filter_side_effect;
+use function Alley\WP\match_block;
+
+global $wp_curate_template_stack;
+
+$wp_curate_template_stack = [];
+
+/**
+ * Registers the block using the metadata loaded from the `block.json` file.
+ * Behind the scenes, it registers also all assets so they can be enqueued
+ * through the block editor in the corresponding context.
+ *
+ * @see https://developer.wordpress.org/reference/functions/register_block_type/
+ */
+function wp_curate_counter_block_init(): void {
+	// Register the block by passing the location of block.json.
+	register_block_type(
+		__DIR__,
+		[
+			'render_callback' => 'wp_curate_render_counter_block',
+		],
+	);
+}
+add_action( 'init', 'wp_curate_counter_block_init' );
+
+/**
+ * Inject a counter block into the post template block being rendered.
+ *
+ * @param WP_Block $parsed_block The block being rendered.
+ * @return WP_Block
+ */
+function wp_curate_inject_counter_block( $parsed_block ): WP_Block {
```

Comment:
I'm getting a fatal error on the frontend because `$parsed_block` is an array.

https://developer.wordpress.org/reference/hooks/render_block_data/

Reply:
same.. will fix

Reply:
there were actually some namespace issues that took care of this

Reply:
@kaitlinbolling this is finally fixed. I finally got phpstan and php runtime to agree.

Reply:
Looks good!